subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+		return sdc_hci_cmd_le_clear_adv_sets();
+#endif
+
+#if defined(CONFIG_BT_OBSERVER)
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_PARAMS:
+		return sdc_hci_cmd_le_set_ext_scan_params((void *)cmd_params);
+
+	case SDC_HCI_OPCODE_CMD_LE_SET_EXT_SCAN_ENABLE:
+		return sdc_hci_cmd_le_set_ext_scan_enable((void *)cmd_params);
+#endif
+
+#if defined(CONFIG_BT_CENTRAL)
+	case SDC_HCI_OPCODE_CMD_LE_EXT_CREATE_CONN:
+		return sdc_hci_cmd_le_ext_create_conn((void *)cmd_params);
+#endif /* CONFIG_BT_CENTRAL */
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+
+	case SDC_HCI_OPCODE_CMD_LE_READ_TRANSMIT_POWER:
+		*param_length_out += sizeof(sdc_hci_cmd_le_read_transmit_power_return_t);
+		return sdc_hci_cmd_le_read_transmit_power((void *)event_out_params);
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	case SDC_HCI_OPCODE_CMD_LE_SET_PRIVACY_MODE:
+		return sdc_hci_cmd_le_set_privacy_mode((void *)cmd_params);
+#endif
+
+	default:
+		return BT_HCI_ERR_UNKNOWN_CMD;
+	}
+}
+
+static uint8_t vs_cmd_put(uint8_t const * const cmd,
+			  uint8_t * const raw_event_out,
+			  uint8_t *param_length_out)
+{
+	uint8_t const *cmd_params = &cmd[BT_HCI_CMD_HDR_SIZE];
+	uint8_t * const event_out_params = &raw_event_out[CMD_COMPLETE_MIN_SIZE];
+	uint16_t opcode = *(uint16_t *)cmd;
+
+	switch (opcode)	{
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_READ_VERSION_INFO:
+		*param_length_out += sizeof(sdc_hci_cmd_vs_zephyr_read_version_info_return_t);
+		return sdc_hci_cmd_vs_zephyr_read_version_info((void *)event_out_params);
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_READ_SUPPORTED_COMMANDS:
+		*param_length_out += sizeof(sdc_hci_cmd_vs_zephyr_read_supported_commands_return_t);
+		return sdc_hci_cmd_vs_zephyr_read_supported_commands((void *)event_out_params);
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_READ_STATIC_ADDRESSES:
+		/* We always return one entry */
+		*param_length_out += sizeof(sdc_hci_cmd_vs_zephyr_read_static_addresses_return_t);
+		*param_length_out += sizeof(sdc_hci_vs_zephyr_static_address_t);
+		return sdc_hci_cmd_vs_zephyr_read_static_addresses((void *)event_out_params);
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_WRITE_TX_POWER:
+		*param_length_out += sizeof(sdc_hci_cmd_vs_zephyr_write_tx_power_return_t);
+		return sdc_hci_cmd_vs_zephyr_write_tx_power((void *)cmd_params,
+							    (void *)event_out_params);
+	case SDC_HCI_OPCODE_CMD_VS_LLPM_MODE_SET:
+		return sdc_hci_cmd_vs_llpm_mode_set((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_CONN_UPDATE:
+		return sdc_hci_cmd_vs_conn_update((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_CONN_EVENT_EXTEND:
+		return sdc_hci_cmd_vs_conn_event_extend((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_QOS_CONN_EVENT_REPORT_ENABLE:
+		return sdc_hci_cmd_vs_qos_conn_event_report_enable((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_EVENT_LENGTH_SET:
+		return sdc_hci_cmd_vs_event_length_set((void *)cmd_params);
+	case SDC_HCI_OPCODE_CMD_VS_ZEPHYR_WRITE_BD_ADDR:
+		return sdc_hci_cmd_vs_zephyr_write_bd_addr((void *)cmd_params);
+	default:
```

Comment:
how about the Non-public APIs?

Reply:
Those are not public, so they are not exposed and used here :)